src/BuildScriptGeneratorCli/Commands/ExecCommand.cs

```diff
@@ -0,0 +1,113 @@
+﻿// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+using System;
+using System.IO;
+using McMaster.Extensions.CommandLineUtils;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Oryx.BuildScriptGenerator;
+using Microsoft.Oryx.Common;
+using Microsoft.Oryx.Common.Extensions;
+
+namespace Microsoft.Oryx.BuildScriptGeneratorCli
+{
+    [Command(Name, Description = "Execute an arbitrary command in the default shell, in an environment " +
+        "with the best-matching platform tool versions.")]
+    internal class ExecCommand : CommandBase
+    {
+        public const string Name = "exec";
+
+        public const string DefaultBashPath = "/bin/bash";
+
+        public const string SrcDirDoesNotExistErrorMessageFmt = "Could not find the source directory '{0}'.";
+        public const string CommandMissingErrorMessage = "A command is required.";
+        public const string NoToolsDetectedErrorMessage = "No usable tools detected for source directory.";
+
+        [Argument(0, Description = "The source directory.")]
+        public string SourceDir { get; set; }
```

Comment:
I would have this is a named parameter, not as positional argument. e.g. `oryx exec -sourceDir /app <command>`

Reply:
and default to current directory

Reply:
I wanted to keep this in-line with `BuildCommand`. WDYT?